cpp/tests/test_odesecir.cpp

```diff
@@ -874,3 +821,72 @@ TEST(Secir, check_constraints_parameters)
     ASSERT_EQ(model.parameters.check_constraints(), 1);
     mio::set_log_level(mio::LogLevel::warn);
 }
+
+TEST(Secir, apply_constraints_parameters)
+{
+    auto model         = mio::osecir::Model(1);
+    auto indx_agegroup = mio::AgeGroup(0);
+    EXPECT_EQ(model.parameters.apply_constraints(), 0);
+
+    mio::set_log_level(mio::LogLevel::off);
+    model.parameters.set<mio::osecir::Seasonality>(-0.5);
+    EXPECT_EQ(model.parameters.apply_constraints(), 1);
+    EXPECT_EQ(model.parameters.get<mio::osecir::Seasonality>(), 0);
+
+    model.parameters.set<mio::osecir::ICUCapacity>(-2);
+    EXPECT_EQ(model.parameters.apply_constraints(), 1);
+    EXPECT_EQ(model.parameters.get<mio::osecir::ICUCapacity>(), 0);
+
+    model.parameters.set<mio::osecir::IncubationTime>(-2);
+    EXPECT_EQ(model.parameters.apply_constraints(), 1);
+    EXPECT_EQ(model.parameters.get<mio::osecir::IncubationTime>()[indx_agegroup], 1e-4);
+
+    model.parameters.set<mio::osecir::SerialInterval>(1);
```

Comment:
This test should set SI=1e-10 to run in the first branch of the `if`, shouldnt it? Also applies to other models